### ISSUE TITLE
can not generate namespace scoped CR from composition.yaml

### ISSUE THREAD

**Issue by haoqing0110** [crossplane/crossplane#1911]
### What happened?

Hello crossplane team, 

I just follow examples in https://github.com/crossplane/crossplane/tree/master/docs/snippets/package to create my own `definition.yaml` and `composition.yaml`. Inside `composition.yaml`, replace RDSInstance with my own resource (a namespace scoped CR).

Then create a `claim.yaml` follow examples in https://github.com/crossplane/crossplane/blob/master/docs/snippets/compose , but failed to generate resource with below error
```
# oc describe composite xxx
...
Events:
  Type     Reason             Age                     From                                                             Message
  ----     ------             ----                    ----                                                             -------
  Normal   SelectComposition  2m50s (x13 over 8m20s)  defined/compositeresourcedefinition.apiextensions.crossplane.io  Successfully selected composition
  Warning  ComposeResources   2m50s (x13 over 8m20s)  defined/compositeresourcedefinition.apiextensions.crossplane.io  cannot compose resource at index 0: cannot apply composed resource: cannot create object: an empty namespace may not be set during creation
```

When I change `composition.yaml` to compose a cluster scoped CR, everything works fine. 

**Is this a bug on namespace scoped CR? or `composition.yaml` doesn't support defining namespace scoped CR?** 

### What environment did it happen in?
Crossplane version: 0.13.0
OpenShift: 4.5.13

**Comment by muvaf** [comment]
Hi @haoqing0110 ! While composition is [currently](https://github.com/crossplane/crossplane/issues/1730) meant to include only managed resources, which are always cluster-scoped, namespaced CR creation should also work to the best of my knowledge. Could you share your `Composition` YAML?

**Comment by haoqing0110** [comment]
@muvaf below are the `Composition` yaml, both meet "an empty namespace may not be set during creation" issue.

Same requirement as https://github.com/crossplane/crossplane/issues/1730#issuecomment-698418882, my original understanding is `Composition` is a compose of any CR, and [Crossplane API conventions](https://crossplane.io/docs/v0.13/introduction/managed-resources.html#syntax) are optional (additional) to original CR. 
```
---
apiVersion: apiextensions.crossplane.io/v1alpha1
kind: Composition
metadata:
  name: compositexxxxinstances.xxx.example.org
  labels:
    provider: xxx
spec:
  writeConnectionSecretsToNamespace: crossplane-system
  compositeTypeRef:
    apiVersion: xxx.example.org/v1alpha1
    kind: CompositeXxxxInstance
  resources:
    - base:
        apiVersion: xxx/v1alpha1
        kind: MyOwnCRD
        spec:
          requests: xxx-operator
          writeConnectionSecretToRef:
            namespace: crossplane-system
      patches:
        - fromFieldPath: "spec.parameters.requests"
          toFieldPath: "spec.requests"
      readinessChecks:
      - fieldPath: status.state
        type: MatchString
        matchString: "Online"
```
```
---
apiVersion: apiextensions.crossplane.io/v1alpha1
kind: Composition
metadata:
  name: compositexxxxinstances.xxx.example.org
  labels:
    provider: xxx
spec:
  writeConnectionSecretsToNamespace: crossplane-system
  compositeTypeRef:
    apiVersion: xxx.example.org/v1alpha1
    kind: CompositeXxxxInstance
  resources:
    - base:
        apiVersion: xxx/v1alpha1
        kind: MyOwnCRD
        metadata:
          namespace: xxx
        spec:
          requests:
          - operands:
            - name: xxx
          writeConnectionSecretToRef:
            namespace: crossplane-system
      patches:
        - fromFieldPath: "spec.parameters.operand"
          toFieldPath: "spec.requests[0].operands[0].name"
        - fromFieldPath: "metadata.namespace"
          toFieldPath: "metadata.namespace"
      readinessChecks:
      - fieldPath: status.state
        type: MatchString
        matchString: "Online"
```

**Comment by Henni** [comment]
Same happens for me on v0.13.0.
Bug might be fixed with v0.15.0 because the code changed quite a bit, but haven't tested it yet.

The closest I got so far: https://github.com/crossplane/crossplane/blob/release-0.13/pkg/controller/apiextensions/composite/reconciler.go#L278

**Comment by muvaf** [comment]
Since https://github.com/crossplane/crossplane/issues/1730 is still up, I think we can close this issue since it's a case we don't aim to cover for now.
